### PR TITLE
AG-8460 Add disableInteractions option on animation manager

### PR DESCRIPTION
**Pull request by lsjroberts**
PR for https://ag-grid.atlassian.net/browse/AG-8460

This adds the `disableInteractions: true` option when animating, which disables/enables interactions while that animation is running. With the intent on queueing some types of interaction to be run when re-enabled in the future.

### CODE REVIEW

File: charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts [hunk 1]
```diff
@@ -176,20 +184,30 @@ export class AnimationManager extends BaseManager<AnimationEventType, AnimationE
...
             return {
                 start: () => {
                     this.updaters.push([id, update]);
                     if (this.requestId == null) {
                         this.startAnimationCycle();
                     }
+
+                    if (disableInteractions) {
+                        this.interactionManager.disable();
+                        this.interactionDisablers.push(id);
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts:198]: Would it make sense to move state/management of tracking who requested disablement into `InteractionManager`? Just wondering if there maybe other use-cases where parts of charts might also want to do this, and if we have queued events they should only be released after all pausers have resumed, without explicitly knowledge of one another?

**Review by alantreadway — COMMENTED** [review]
Generally happy with this, but I'm just wondering about the point mentioned?

File: charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts (hunk 1 above)
Comment (lsjroberts) [reply charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts:198]: Good shout, yeah that would make more sense. So I guess `interactionManager.disable(id)` and then `.enable(id)`. But given `.enable(id)` wouldn't necessarily actual enable it, it probably needs a different name like your saying with pause and... something

Comment (lsjroberts) [reply charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts]: Went with `disable(id)` and `removeDisable(id)`. Thoughts?

Comment (lsjroberts) [reply charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts]: Went with `pause(id)` and `resume(id)`

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**